scripts/build/tools/ebs_snapshot.py

```diff
@@ -0,0 +1,167 @@
+                    failure += 1
+
+    return success, failure
+
+def list_snapshot(tag_keyvalue, snap_description):
+    """
+    Find all EBS snapshots that have a matching tag value AND description. Prints snap id, description, tags, and start time.
+
+    :param tag_keyvalue: List of Strings with tag keyvalues in the form of "key:value"
+    :param snap_description: String with the snapshot description to search
+    :return: None
+    """
+
+    tag_filter = []
+    description_filter = {"Name": "description", "Values": [snap_description]}
+    ec2_client = boto3.resource('ec2')
+    
+    for keyvalue in tag_keyvalue:
+        tag_filter.append(_kv_to_dict(keyvalue))
+
+    for tags in tag_filter:
+        response = ec2_client.snapshots.filter(Filters=[tags,description_filter])
+        log.info(f'Getting snapshots with tags that match {tags}...')
+        num_snaps = len(list(response))
+        log.info(f'Tag {tags} has {num_snaps} snapshots')
+        snap_list = sorted(response, key=lambda k: k.start_time)
+        for n in range(num_snaps):
+            print(f'Snap ID: {snap_list[n].snapshot_id} \n Description: {snap_list[n].description} \n Tags: {snap_list[n].tags} \n Start Time: {snap_list[n].start_time}')
+
+    return None
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Script to manage EBS snapshots for build artifacts')
+    parser.add_argument('--action', '-a', type=str, help='(create|delete|list) Creates, deletes, or lists EBS snapshots based on tag. Requires --tags argument')
+    parser.add_argument('--tags', '-t', type=str, required=True, help='Comma separated key value tags to search for in the form of "key:value", for example, "PipelineAndBranch:default_development","PipelineAndBranch:default_development"')
+    parser.add_argument('--description', '-d', default=DEFAULT_SNAPSHOT_DESCRIPTION, help=f'Snapshot description to write or search for. Defaults to "{DEFAULT_SNAPSHOT_DESCRIPTION}"')
+    parser.add_argument('--retention', '-r', default=DEFAULT_SNAPSHOT_RETAIN, type=int, help=f'Integer with the number of snapshots to retain. Defaults to {DEFAULT_SNAPSHOT_RETAIN}')
+    parser.add_argument('--execute', '-e', action='store_false', help=f'Execute the snapshot commands. This needs to be set, otherwise it will always dryrun')
+    return parser.parse_args()
+
+def main():
+    args = parse_args()
+    tag_list = args.tags.split(",")
+    print(tag_list)
+    if 'create' in args.action:
+        ret = create_snapshot(tag_list, args.description, args.execute)
+        log.info(f'{ret[0]} snapshots created, {ret[1]} snapshots failed')
+    elif 'delete' in args.action:
+        ret = delete_snapshot(tag_list, args.description, args.retention, args.execute)
+        log.info(f'{ret[0]} snapshots deleted, {ret[1]} snapshot deletions failed')
+    elif 'list' in args.action:
+        ret = list_snapshot(tag_list, args.description)
+
+if __name__ == "__main__":
+    sys.exit(main())
```

Comment:
minor, add a newline at end of file for Linux

Reply:
Yup, will change